src/backend/common/util.hpp

```diff
@@ -21,3 +21,21 @@ void saveKernel(const std::string& funcName, const std::string& jit_ker,
                 const std::string& ext);
 
 std::string int_version_to_string(int version);
+
+const std::string& getCacheDirectory();
+
+bool directoryExists(const std::string& path);
+
+bool createDirectory(const std::string& path);
+
+bool removeFile(const std::string& path);
+
+bool renameFile(const std::string& sourcePath, const std::string& destPath);
+
+bool isDirectoryWritable(const std::string& path);
+
+std::string makeTempFilename();
+
+std::size_t deterministicHash(const void* data, std::size_t byteSize);
+
+std::size_t deterministicHash(const std::string& data);
```

Comment:
All directory helpers are self-explanatory with their name. The following however would make understanding easier if they had one or two line documentation above them.
- `makeTempFilename` - something like what to expect in the output string as the name is also a little vague. Function says file name, but in actual it is returning just a string based on active thread ID and current callee counter value in a given thread. I understand the intent, but a couple of lines of docs above it and perhaps a name change could help for future developers.
- `deterministicHash` - You could say in docs that this fn currently return FNV1a hash
- `deterministicHash` - this version is just a wrapper around the other one.

Reply:
I have added the documentation, please let me know if it is adequate!

Reply:
Looks good.